lib/puppet/provider/cisco_bridge_domain/cisco.rb

```diff
@@ -0,0 +1,147 @@
+# The NXAPI provider for cisco bridge domain.
+#
+# March, 2016
+#
+# Copyright (c) 2014-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'autogen.rb'))
+end
+
+Puppet::Type.type(:cisco_bridge_domain).provide(:cisco) do
+  desc 'The new NXAPI provider.'
+
+  confine feature: :cisco_node_utils
+  defaultfor operatingsystem: :nexus
+
+  mk_resource_methods
+
+  BD_NON_BOOL_PROPS = [:bd_name]
+  BD_BOOL_PROPS = [:fabric_control, :shutdown]
+  BD_ALL_PROPS = BD_NON_BOOL_PROPS + BD_BOOL_PROPS
+
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@bd',
+                                            BD_NON_BOOL_PROPS)
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:bool, self, '@bd',
+                                            BD_BOOL_PROPS)
+
+  def initialize(value={})
+    super(value)
+    @bd = Cisco::BridgeDomain.bds[@property_hash[:name]]
+    @property_flush = {}
+    debug 'Created provider instance of cisco_bridge_domain.'
+  end
+
+  def self.properties_get(bd_id, v)
```

Comment:
Nit: This is okay but consider using the naming convention Mike and I have been using, which is a little clearer as far as indicating that these are node utils references, ie: 

```
-  def self.properties_get(bd_id, v)
+  def self.properties_get(bd_id, nu_obj)
``` 

 and

```
-  PuppetX::Cisco::AutoGen.mk_puppet_methods(:bool, self, '@bd',
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:bool, self, '@nu',
```

...but I won't make you change it. :-)